src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt

```diff
@@ -46,13 +93,19 @@ class TransactionsRefundQueueConsumer(
 
   var logger: Logger = LoggerFactory.getLogger(TransactionsRefundQueueConsumer::class.java)
 
-  private fun getTransactionIdFromPayload(data: BinaryData): Mono<String> {
-    val idFromActivatedEvent =
-      data.toObjectAsync(TransactionExpiredEvent::class.java).map { it.transactionId }
-    val idFromRefundedEvent =
-      data.toObjectAsync(TransactionRefundRetriedEvent::class.java).map { it.transactionId }
+  private fun parseEvent(data: BinaryData): Mono<RefundQueueEvent> {
+    val refundRequestedEvent = data.toObjectAsync(TransactionRefundRequestedEvent::class.java)
+    val refundRetriedEvent = data.toObjectAsync(TransactionRefundRetriedEvent::class.java)
+    val expiredEvent = data.toObjectAsync(TransactionExpiredEvent::class.java)
```

Comment:
I think that TransactionExpiredEvent should not be sent on refund queue, right?

Reply:
You're right, removing it

Reply:
Fixed in 32d29685af54e9875fd29620a6e07ea8dc76c453

Reply:
You're right, fixed in 32d2968